s3/src/test/scala/uk/gov/nationalarchives/DAS3ClientTest.scala

```diff
@@ -223,6 +215,60 @@ class DAS3ClientTest extends AnyFlatSpec with MockitoSugar {
     ex.getMessage should equal("Error calling delete objects")
   }
 
+  "listKeysWithCommonPrefixes" should "return a publisher with the expected type" in {
+    // You can't mock/spy on the publisher properly due to the commonPrefix method being final,
+    // so this test just makes sure that method returns a SdkPublisher[String] in case
+
+    val asyncClientMock = mock[S3AsyncClient]
+
+    val listObjectsV2Request = ListObjectsV2Request.builder
+      .bucket("bucket")
+      .delimiter("/")
+      .prefix("prefix")
+      .build
+
+    val listObjectsV2Publisher = new ListObjectsV2Publisher(asyncClientMock, listObjectsV2Request)
+
+    when(asyncClientMock.listObjectsV2Paginator(any[ListObjectsV2Request])).thenReturn(listObjectsV2Publisher)
+    val client = DAS3Client[IO](asyncClientMock)
+    val publisher: SdkPublisher[String] = client.listKeysWithCommonPrefixes("bucket", "keyPrefix/").unsafeRunSync()
+  }
+
+  "listKeysWithCommonPrefixes" should "make a request to 'listKeysWithCommonPrefixes' with the correct arguments" in {
+    val listObjectsV2PaginatorCaptor: ArgumentCaptor[ListObjectsV2Request] =
+      ArgumentCaptor.forClass(classOf[ListObjectsV2Request])
+    val asyncClientMock = mock[S3AsyncClient]
+    val listObjectsV2Request = ListObjectsV2Request.builder
+      .bucket("bucket")
+      .delimiter("/")
+      .prefix("prefix")
+      .build
+
+    val listObjectsV2Publisher = new ListObjectsV2Publisher(asyncClientMock, listObjectsV2Request)
+    when(asyncClientMock.listObjectsV2Paginator(listObjectsV2PaginatorCaptor.capture()))
+      .thenReturn(listObjectsV2Publisher)
+    val client = DAS3Client[IO](asyncClientMock)
+    client.listKeysWithCommonPrefixes("bucket", "keyPrefix/").unsafeRunSync()
+
+    val listObjectsRequest = listObjectsV2PaginatorCaptor.getValue
+    listObjectsRequest.delimiter() should equal("/")
+    listObjectsRequest.prefix() should equal("keyPrefix/")
+    listObjectsRequest.bucket() should equal("bucket")
+  }
+
+  "listKeysWithCommonPrefixes" should "return an error if the transfer manager returns an error" in {
```

Comment:
Minor thing but listKeysWithCommonPrefixes is using the async client, not the transfer manager.

Reply:
🤦 ah, good spot; forgot to rename this.